### ISSUE TITLE
grid search with cv should have development/evaluation sets

### ISSUE THREAD

**Issue by mobeets** [mobeets/mASD#15]
see [here](http://scikit-learn.org/stable/modules/cross_validation.html)

What I am currently doing:
- grid hyperparameter space
- for each fold, find the hyperparameter/fit that maximizes the test set
- report all scores/hypers, and plot the first one (or something like that)

What I should be doing:
- split 80%/20% for development/evaluation set
- do the first two steps above on the development set
- using the hyper with maximum score (?), evaluate performance and fit on the the evaluation set

**Comment by mobeets** [comment]
in `reg.cvMaxScoreGrid` and `reg.cvMaxScoreGridSearch`, before and after fitting.

**Comment by mobeets** [comment]
Note that splitting 80%/20% leaves the development set with fewer trials than it has parameters, so fits get janky. Instead I am just going to do a 50%/50% split for dev/evaluation, but the hyperparameter choosing will be done on the entire 100% so as not to throw away trials. Better than nothing...
